server/server/src/account_service.rs

```diff
@@ -167,3 +177,22 @@ pub async fn delete_account(
 
     Ok(())
 }
+
+pub fn is_user_authorized<T: Debug>(
+    public_key: &PublicKey, admins: &HashSet<Username>, db: &ServerV1,
+) -> Result<bool, ServerError<T>> {
+    let mut is_authorized = false;
+
+    for admin_username in admins {
+        let admin = db.usernames.get(admin_username)?.ok_or_else(|| {
+            internal!("An admin does not exist in db, username: {}", admin_username)
+        })?;
+
+        if admin.0 == *public_key {
+            is_authorized = true;
+            break;
+        }
+    }
+
+    Ok(is_authorized)
+}
```

Comment:
There should be no loop involved in this calculation:
1. lookup the public key's username
2. check if that username is an admin